app/src/main/cpp/src/components/transport_manager/src/bluetooth_le/ble_control_protocol.cc

```diff
@@ -0,0 +1,107 @@
+
+namespace transport_manager {
+namespace transport_adapter {
+
+SDL_CREATE_LOG_VARIABLE("TransportManager")
+
+    namespace {
+        std::map<BleProtolActions, std::string> mapActionNames = {
+        {BleProtolActions::ON_DEVICE_CONNECTED, "ON_DEVICE_CONNECTED"},
+        {BleProtolActions::ON_DEVICE_DISCONNECTED, "ON_DEVICE_DISCONNECTED"}};
+
+        const char cDefaultValue[] = "_";
+        const char cActionKey[] = "action";
+        const char cNameKey[] = "name";
+        const char cAddressKey[] = "address";
+        const char cParamsKey[] = "params";
+    }
+
+    BleProtolActions BleControlProtocol::GetMessageActionType(const MsgData& message) {
+
+        utils::JsonReader reader_;
+        Json::Value value;
+        
+        if(reader_.parse((char*)(message.data()), &value)) {
+
+            auto action = value.get(cActionKey, cDefaultValue);
+
+            if(action.asString() != cDefaultValue) {
+                return GetActionFromString(action.asString());
+            }
+
+        }
+
+        return BleProtolActions::INVALID_ENUM;
+    }
+
+    BleProtolActions BleControlProtocol::GetActionFromString(const std::string& str) {
+        
+        BleProtolActions result = BleProtolActions::INVALID_ENUM;
+
+        auto findResult = std::find_if(mapActionNames.begin(), mapActionNames.end(),
+         [&](const std::pair<BleProtolActions, std::string> &pair) {
+            return pair.second == str;});
+
+        if(findResult != mapActionNames.end()) {
+            result = findResult->first;
+        }    
+
+        return result;
+    }
+
+    std::string BleControlProtocol::GetActionString(const BleProtolActions action) {
+        auto it = mapActionNames.find(action);
+        return it != mapActionNames.end() ? it->second : "UNKNOWN";
+    }
+
+    std::string BleControlProtocol::GetAddress(const MsgData& message) {
+
+        utils::JsonReader reader_;
+        Json::Value value;
+        
+        if(reader_.parse((char*)(message.data()), &value)) {
+
+            auto params = value.get(cParamsKey, Json::Value::null);
+
+            if(params != Json::Value::null) {
+
+                auto addr = params.get(cAddressKey, cDefaultValue);
+                if(addr.asString() != cDefaultValue) {
+                    return addr.asString();
+                }
+            }
+        }
+
+        return std::string();
+    }
+    
+    std::string BleControlProtocol::GetName(const MsgData& message) {
+        utils::JsonReader reader_;
+        Json::Value value;
+        
+        if(reader_.parse((char*)(message.data()), &value)) {
+
+            auto params = value.get(cParamsKey, Json::Value::null);
+
+            if(params != Json::Value::null) {
+
+                auto name = params.get(cNameKey, cDefaultValue);
+                if(name.asString() != cDefaultValue) {
+                    return name.asString();
+                }
+            }
+        }
+
+        return std::string();
+    }
+
+}
+}
```

Comment:
@VadymLuchko add an empty line

Reply:
fixed in 22caff304f02304d55c097802b850d8bca25a370